### PR TITLE
Replaceing KV actor by Ray API

### LINKED ISSUE #135: [Multi-Job] The KV Actor increase management burdens of Multi-Job
Client mode's internal_kv is implemented by creating a named Actor for transporting kv data between Ray cluster and client driver.

This implementation:
1. forces the driver to manage its lifecycle, e.g. kill it when fed.shutdown;
2. In multi-job mode, each Job has to maintain its own KV actor with a unique actor name.

However, none of the above are unnecessary because Ray already provides an official interface for KV accessing under client mode, no Actor is needed.

Replacing KV Actor to the official interface

### PR DESCRIPTION
**Pull request by NKcqx**
Resolves #135

### CODE REVIEW

File: fed/_private/compatible_utils.py [hunk 1]
```diff
@@ -128,39 +128,33 @@ class ClientModeInternalKv(AbstractInternalKv):
...
         super().__init__()
-        self._internal_kv_actor = ray.get_actor("_INTERNAL_KV_ACTOR")
+        self._client_api_stub = ray.util.client.ray
 
     def initialize(self):
-        o = self._internal_kv_actor.initialize.remote()
-        return ray.get(o)
+        # Note(NKcqx): internval_kv is always initiated after `ray.init`,
+        # calling this is equal to directly return "True"
+        return self._client_api_stub._internal_kv_initialized()
```
Comment (jovany-wang) [inline fed/_private/compatible_utils.py:136]: So we should mark this method as deprecated?

File: fed/_private/compatible_utils.py [hunk 2]
```diff
@@ -128,39 +128,33 @@ class ClientModeInternalKv(AbstractInternalKv):
...
 
     def delete(self, k):
-        o = self._internal_kv_actor.delete.remote(k)
-        return ray.get(o)
+        return self._client_api_stub._internal_kv_del(k)
 
     def reset(self):
-        o = self._internal_kv_actor.reset.remote()
-        return ray.get(o)
+        # Note(NKcqx): No `gcs_client` is instantiated for kv, and the 'initialized'
```
Comment (jovany-wang) [inline fed/_private/compatible_utils.py:148]: Mark as deprecated.

**Review by jovany-wang — COMMENTED** [review]
As discussed offline, we should enable tests in client mode.

**Comment by NKcqx** [comment]
Unfortunately, Ray's client mode internal_kv accessing has a BUG that causes the call of `internal_kv_del` takes no effect. This PR can't be merged untill this [issue](https://github.com/ray-project/ray/issues/36945) is fixed.

**Comment by NKcqx** [comment]
It turns out to be a dumb BUG in Ray, fix PR: 
https://github.com/ray-project/ray/pull/37031/files

**Comment by jovany-wang** [comment]
If the current ray doesn't support it, how can we handle to get compatible with all versions?

**Comment by jovany-wang** [comment]
Do we need go with this PR? @ray-project/rayfed

**Review by jovany-wang — APPROVED** [review]
LGTM

**Pull request merged**